src/Microsoft.AspNet.JsonPatch/Helpers/ExpandoObjectDictionaryExtensions.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.JsonPatch.Helpers
+{
+    // Helper methods to allow case-insensitive key search
+    internal static class ExpandoObjectDictionaryExtensions
+    {
+        internal static void SetValueForCaseInsensitiveKey(this IDictionary<string, object> propertyDictionary,
+        string key, object value)
+        {
+            foreach (KeyValuePair<string, object> kvp in propertyDictionary)
+            {
+                if (string.Equals(kvp.Key, key, StringComparison.OrdinalIgnoreCase))
+                {
+                    propertyDictionary[kvp.Key] = value;
+                    break;
+                }
+            }
+        }
+
+        internal static void RemoveValueForCaseInsensitiveKey(this IDictionary<string, object> propertyDictionary,
+        string key)
+        {
+            string realKey = "";
+            foreach (KeyValuePair<string, object> kvp in propertyDictionary)
+            {
+                if (string.Equals(kvp.Key, key, StringComparison.OrdinalIgnoreCase))
+                {
+                    realKey = kvp.Key;
+                    break;
+                }
+            }
+
+            if (realKey != "")
+            {
+                propertyDictionary.Remove(realKey);
+            }
+        }
```

Comment:
What should we do if the dictionary has the same key with multiple casings? Would this ever happen?